src/single-spa-svelte.js

```diff
@@ -46,21 +45,56 @@ function mount(opts, mountedInstances) {
     return object;
   }, {});
 
-  return new Promise((resolve, reject) => {
+  return Promise.resolve().then(() => {
+    const domElementGetter = chooseDomElementGetter(opts, singleSpaProps);
+    const domElement = domElementGetter();
+    // See https://svelte.dev/docs#Creating_a_component
     mountedInstances.instance = new opts.component({
       ...svelteOpts,
-      target: opts.domElementGetter(),
-      data: opts.data || {},
+      target: domElement,
+      props: Object.assign({}, singleSpaProps, opts.props)
```

Comment:
According to latest svelte docs, data should be passed in as `props`, not `data` (I think). See https://svelte.dev/docs#Client-side_component_API

Due to this change, I plan on releasing this PR as a new major version